common/src/main/java/com/msd/gin/halyard/common/HalyardTableUtils.java

```diff
@@ -401,8 +403,17 @@ public static Statement parseStatement(Cell c) {
         bb.get(ob);
         byte[] cb = new byte[bb.remaining()];
         bb.get(cb);
-        ValueFactory vf = SimpleValueFactory.getInstance();
-        return vf.createStatement(NTriplesUtil.parseResource(new String(sb, UTF8), vf), NTriplesUtil.parseURI(new String(pb, UTF8), vf), NTriplesUtil.parseValue(new String(ob,UTF8), vf), cb.length == 0 ? null : NTriplesUtil.parseResource(new String(cb,UTF8), vf));
+        Resource subj = NTriplesUtil.parseResource(new String(sb, UTF8), vf);
+        IRI pred = NTriplesUtil.parseURI(new String(pb, UTF8), vf);
+        Value value = NTriplesUtil.parseValue(new String(ob,UTF8), vf);
+		Statement stmt;
+		if (cb.length == 0) {
+			stmt = vf.createStatement(subj, pred, value);
```

Comment:
NB, this 3-arg method is designed to return a different Statement impl than the 4-arg method to give a ~20% saving in memory (important consideration for massive group-bys).